node/narwhal/src/event/batch_signature.rs

```diff
@@ -55,3 +55,38 @@ impl<N: Network> EventTrait for BatchSignature<N> {
         })
     }
 }
+
+#[cfg(test)]
+pub mod prop_tests {
+    use crate::{
+        event::{
+            certificate_request::prop_tests::any_field,
+            challenge_response::prop_tests::any_signature,
+            EventTrait,
+        },
+        helpers::now,
+        BatchSignature,
+    };
+    use bytes::{BufMut, BytesMut};
+    use proptest::prelude::{BoxedStrategy, Just, Strategy};
+    use test_strategy::proptest;
+
+    type CurrentNetwork = snarkvm::prelude::Testnet3;
+
+    pub fn any_batch_signature() -> BoxedStrategy<BatchSignature<CurrentNetwork>> {
+        (any_field(), any_signature(), Just(now()), -10..10i64)
```

Comment:
is this the only drift range that we accept? seems a bit narrow at first glance

Reply:
Perhaps, however none of the code in `src/event` does anything with those timestamps. Once we expand proptesting to `Gateway::inbound` and other call sites, we can generate additional strategies for wider ranges.